icu4c/source/common/caniter.cpp

```diff
@@ -490,6 +493,15 @@ Hashtable *CanonicalIterator::getEquivalents2(Hashtable *fillinResult, const cha
 
                 ne = remainder.nextElement(el);
             }
+            // ICU-22642 To avoid security attack by an useless string which
+            // have too many permutations to hang the function, limit the
```

Comment:
I assume you've mistyped here, this guards against strings that have so many permutations that they would otherwise hang the function and there's no such thing as having too many permutations to hang the function, right?